### PR TITLE
revolution: disable servo channel when used for LED

### PR DESCRIPTION
**Pull request by mlyle**

### CODE REVIEW

**Comment by mlyle** [comment]
jenkins, build artifacts please

File: flight/PiOS/STM32/pios_servo.c [hunk 1]
```diff
@@ -54,6 +54,8 @@
 /* Private variables */
 static const struct pios_servo_cfg *servo_cfg;
 
+static int channel_mask;
```
Comment (tracernz) [inline flight/PiOS/STM32/pios_servo.c]: Would prefer an explicit initializer for clarity but meh. unsigned too imo.

File: flight/PiOS/STM32/pios_servo.c [hunk 2]
```diff
@@ -153,20 +155,25 @@ int32_t PIOS_Servo_Init(const struct pios_servo_cfg *cfg)
...
-	for (i = 0; i < max_dio; i++) {
-		if (i >= servo_cfg->num_channels) {
-			break;
+	for (int i = 0; (i < servo_cfg->num_channels) && (out < max_dio); i++) {
+		if (channel_mask & (1 << i)) {
+			continue;
 		}
 
-		dios[i] = DIO_MAKE_TAG(servo_cfg->channels[i].pin.gpio,
+		dios[out++] = DIO_MAKE_TAG(servo_cfg->channels[i].pin.gpio,
```
Comment (tracernz) [inline flight/PiOS/STM32/pios_servo.c]: This is a bit messy from user perspective, BLHeli number motoring doesn't match dR. :/ Should really document the function properly and note that numbering is munched so it doesn't bite somebody.

Comment (mlyle) [reply flight/PiOS/STM32/pios_servo.c]: Or define a DIO_NULL

**Comment by mlyle** [comment]
OK, took the slightly riskier and less lazy (but more correct) approach :P

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
test feedback wanted-- amstan or anyone...

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
Claimed not to work--deferring

**Comment by mlyle** [comment]
Mind changed.. undeferring.. :P

**Comment by mlyle** [comment]
Claimed to work, but needs really careful review.

**Comment by mlyle** [comment]
Probable that this gets deferred because it's scary and it's late and the test feedback is ambiguous

**Pull request merged**